helm-chart/templates/deployment.yaml

```diff
@@ -43,22 +51,39 @@ spec:
             claimName: itop-env-pvc
         - name: itop-log-pv
           persistentVolumeClaim:
-            claimName: itop-log-pvc   
+            claimName: itop-log-pvc
+        - name: itop-extensions-pv
+          persistentVolumeClaim:
+            claimName: itop-extensions-pvc
+      {{- end }}
+      {{- end }}
       containers:
         - name: {{ .Chart.Name }}
           securityContext:
             {{- toYaml .Values.securityContext | nindent 12 }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           volumeMounts:
+            {{- if .Values.pvc.enabled }}
             - mountPath: "/var/www/html/data"
-              name: itop-data-pv
+              name: {{ ternary "itop-pv" "itop-data-pv" .Values.pvc.useOnePV }}
+              subPath: {{ ternary "data" "" .Values.pvc.useOnePV }}
             - mountPath: "/var/www/html/env-production"
-              name: itop-env-pv
+              name: {{ ternary "itop-pv" "itop-env-pv" .Values.pvc.useOnePV }}
+              subPath: {{ ternary "env-production" "" .Values.pvc.useOnePV }}
             - mountPath: "/var/www/html/log"
-              name: itop-log-pv
+              name: {{ ternary "itop-pv" "itop-log-pv" .Values.pvc.useOnePV }}
+              subPath: {{ ternary "log" "" .Values.pvc.useOnePV }}
             - mountPath: "/var/www/html/conf"
-              name: itop-conf-pv
+              name: {{ ternary "itop-pv" "itop-conf-pv" .Values.pvc.useOnePV }}
+              subPath: {{ ternary "conf" "" .Values.pvc.useOnePV }}
+            - mountPath: "/var/www/html/extensions"
+              name: {{ ternary "itop-pv" "itop-extensions-pv" .Values.pvc.useOnePV }}
+              subPath: {{ ternary "extensions" "" .Values.pvc.useOnePV }}
```

Comment:
I'm wondering if we need to add this as a volume. Shouldn't the `.Values.extraVolumes` be used for mounting extensions, if needed?

Reply:
I think we should keep "extensions" folder for persistence data to avoid frustration.(https://www.itophub.io/wiki/page?id=latest:install:upgrading_itop)

extraVolumes can be used for mount certs for MySql SSL purpose.

Reply:
I don't see how this link is relevant? Usually for extensions we don't use a persistent data volume, but a mounted volume. Since there isn't anything to persist..

Reply:
from link:
5. Your iTop directory should now contains
    the following subdirectories : conf, data, env-production, log, extensions, web (and maybe toolkit or collectors if you have installed them)
    
and by my logic, while upgrade itop needs for extensions. where the app will be able to get files if folder not persistence and used some extensions?

Reply:
I suggest a compromise solution: to add option that disable\enable this volume for persistence.

Reply:
Okay, that's a good compromise for me.

Reply:
done